app/controllers/api/v1/experiences_controller.rb

```diff
@@ -62,20 +47,112 @@ def destroy
     private
       def set_user
         @user = User.find(params[:user_id])
+
+        if @user.id != current_user.id
+          raise ExceptionTypes::UnauthorizedError.new("You do not have access to modify the experiences of the user with ID #{@user.id}")
+        end
       end
 
-      # Use callbacks to share common setup or constraints between actions.
       def set_experience
         @experience = @user.experiences.find(params[:id]) if @user
       end
 
-      # Refactor these to be one method that takes a parameter for appropriate ID
-      def organization_experience_params
-        params.require(:experience).permit(:user_id, :organization_id, :start_date, :end_date, :title, :award, :current)
+      # Validate the request payload for creating a new experience
+      def validate_create_params
```

Comment:
So is the main purpose of the validation functions to generate exceptions if something goes wrong? (Since it doesn't look like we use any of the return values.)

Reply:
Good question. Yes, the main purpose of these validate functions are to check for erroneous conditions that can't be easily recovered from. If they pass, then the actual action method (in this case either create or update) will execute normally and confidently pass the params to the model method that may or may not validate further aspects of each param.